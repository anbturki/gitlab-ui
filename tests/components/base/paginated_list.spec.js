import { shallowMount } from '@vue/test-utils';
import PaginatedList from '../../../src/components/base/paginated_list/paginated_list.vue';
import GlSearchBoxByType from '../../../src/components/base/search_box_by_type/search_box_by_type.vue';
import GlPagination from '../../../src/components/base/pagination/pagination.vue';

describe('Paginated List', () => {
  let wrapper;
  let template;

  beforeAll(() => {
    template = `
      <div class="slot" slot-scope="{ listItem }">
        <span class="item">Item Name: {{listItem.id}}</span>
      </div>
    `;
  });

  afterEach(() => {
    wrapper.destroy();

    wrapper = null;
  });

  const createComponent = (props = {}) => {
    wrapper = shallowMount(PaginatedList, {
      scopedSlots: {
        default: template,
      },
      propsData: {
        list: [],
        ...props,
      },
      sync: false,
    });
  };

  describe('Searchless states', () => {
    it('renders the list in an empty state, when no items are provided', () => {
      createComponent();

      expect(wrapper.findAll('.item').length).toEqual(0);
      expect(wrapper.findAll('.empty-message').length).toEqual(1);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the list in an empty state with a custom empty message, when one is provided', () => {
      const msg = 'List is passed in empty';

      createComponent({ emptyMessage: msg });

      expect(wrapper.findAll('.item').length).toEqual(0);
      expect(wrapper.find('.empty-message').text()).toEqual(msg);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the list with nested items, when an array of items is provided', () => {
      createComponent({
        list: [{ id: 'foo' }, { id: 'bar' }],
      });

      expect(wrapper.findAll('.item').length).toEqual(2);
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('Search states', () => {
    const list = [{ id: 'foo' }, { id: 'bar' }, { id: 'baz' }];

    it('renders the list flitered by search results', () => {
      createComponent({ list });

      const search = wrapper.find(GlSearchBoxByType);
      search.vm.$emit('input', 'ba');

      return wrapper.vm.$nextTick().then(() => {
        expect(wrapper.findAll('.item').length).toEqual(2);
        expect(wrapper.element).toMatchSnapshot();
      });
    });

    it('renders the default empty search message when there are no search results', () => {
      createComponent({ list });

      const search = wrapper.find(GlSearchBoxByType);
      search.vm.$emit('input', 'qux');

      return wrapper.vm.$nextTick().then(() => {
        expect(wrapper.findAll('.item').length).toEqual(0);
        expect(wrapper.findAll('.empty-search').length).toEqual(1);
        expect(wrapper.element).toMatchSnapshot();
      });
    });

    it('renders the custom empty search message when there are no search results', () => {
      const msg = 'Custom empty message';

      createComponent({ list, emptySearchMessage: msg });

      const search = wrapper.find(GlSearchBoxByType);
      search.vm.$emit('input', 'qux');

      return wrapper.vm.$nextTick().then(() => {
        expect(wrapper.findAll('.item').length).toEqual(0);
        expect(wrapper.find('.empty-search').text()).toEqual(msg);
        expect(wrapper.element).toMatchSnapshot();
      });
    });
  });

  describe('Pagination', () => {
    const list = [
      { id: 'foo' },
      { id: 'bar' },
      { id: 'baz' },
      { id: 'qux' },
      { id: 'quux' },
      { id: 'corge' },
      { id: 'grault' },
      { id: 'garply' },
      { id: 'waldo' },
      { id: 'fred' },
      { id: 'xyzzy' },
      { id: 'plugh' },
      { id: 'thud' },
    ];

    it('renders 10 items for a default page size of 10, 13 total', () => {
      createComponent({ list });

      expect(wrapper.findAll('.item').length).toEqual(10);
      expect(wrapper.find(GlPagination).props('totalItems')).toEqual(13);
      expect(wrapper.find(GlPagination).props('perPage')).toEqual(10);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders 13 items for a default page size of 20', () => {
      createComponent({ list, perPage: 20 });

      expect(wrapper.findAll('.item').length).toEqual(13);
      expect(wrapper.find(GlPagination).props('totalItems')).toEqual(13);
      expect(wrapper.find(GlPagination).props('perPage')).toEqual(20);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders 5 items on page 1 for a page size of 5, 13 total itmes and renders pagination', () => {
      createComponent({ list, perPage: 5 });

      expect(wrapper.findAll('.item').length).toEqual(5);
      expect(wrapper.find(GlPagination).props('totalItems')).toEqual(13);
      expect(wrapper.find(GlPagination).props('perPage')).toEqual(5);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders 3 items on page 2 with default page size of 10 and 13 total items', () => {
      createComponent({ list, page: 2 });

      expect(wrapper.findAll('.item').length).toEqual(3);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders 1 item on page 7 with page size of 2 and 13 total items', () => {
      createComponent({ list, page: 7, perPage: 2 });

      expect(wrapper.findAll('.item').length).toEqual(1);
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('props', () => {
    describe('filterable', () => {
      it('has search enabled by default', () => {
        createComponent();

        expect(wrapper.find(GlSearchBoxByType).exists()).toBeTruthy();
      });

      it('has search disabled when filterable prop set to false', () => {
        createComponent({ filterable: false });

        expect(wrapper.findAll('[aria-label="Search"]').length).toEqual(0);
      });
    });

    describe('filter', () => {
      const list = [{ id: 'foo' }, { id: 'bar' }];

      it('filters on default "id" key', () => {
        createComponent({ list });

        wrapper.setData({ queryStr: 'ba' });

        return wrapper.vm.$nextTick().then(() => {
          expect(wrapper.findAll('.item').length).toEqual(1);
          expect(wrapper.element).toMatchSnapshot();
        });
      });

      it('filters on provided "myKey" key', () => {
        createComponent({ list: [{ myKey: 'foo' }, { myKey: 'bar' }], filter: 'myKey' });

        wrapper.setData({ queryStr: 'ba' });

        return wrapper.vm.$nextTick().then(() => {
          expect(wrapper.findAll('.item').length).toEqual(1);
          expect(wrapper.element).toMatchSnapshot();
        });
      });

      it('filters with provided filter function', () => {
        const filter = (item, queryStr) => item.id.includes(queryStr);

        createComponent({ list, filter });

        wrapper.setData({ queryStr: 'ba' });

        return wrapper.vm.$nextTick().then(() => {
          const elem = wrapper.findAll('.item');
          expect(elem.length).toEqual(1);
          expect(elem.at(0).text()).toBe('Item Name: bar');
          expect(wrapper.element).toMatchSnapshot();
        });
      });
    });
  });
});
