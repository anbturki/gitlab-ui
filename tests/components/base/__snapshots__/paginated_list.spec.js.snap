// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginated List Pagination renders 1 item on page 7 with page size of 2 and 13 total items 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between flex-row-reverse"
  >
     
    <gl-search-box-by-type-stub
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: thud
        </span>
      </div>
    </li>
  </ul>
   
  <gl-pagination-stub
    align="left"
    class="d-flex justify-content-center prepend-top-default"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnextpage="Go to next page"
    labelpage="page => \`Go to page \${page}\`"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next ›"
    perpage="2"
    prevtext="‹ Prev"
    totalitems="13"
    value="7"
  />
   
  <!---->
</div>
`;

exports[`Paginated List Pagination renders 3 items on page 2 with default page size of 10 and 13 total items 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between flex-row-reverse"
  >
     
    <gl-search-box-by-type-stub
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: xyzzy
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: plugh
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: thud
        </span>
      </div>
    </li>
  </ul>
   
  <gl-pagination-stub
    align="left"
    class="d-flex justify-content-center prepend-top-default"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnextpage="Go to next page"
    labelpage="page => \`Go to page \${page}\`"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next ›"
    perpage="10"
    prevtext="‹ Prev"
    totalitems="13"
    value="2"
  />
   
  <!---->
</div>
`;

exports[`Paginated List Pagination renders 5 items on page 1 for a page size of 5, 13 total itmes and renders pagination 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between flex-row-reverse"
  >
     
    <div
      class="position-relative ms-no-clear"
    >
      <input
        aria-label="Search"
        class="pl-5 form-control"
        placeholder="Search"
        type="text"
      />
       
      <div
        class="pl-2 pr-2 position-left-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
      >
        <i
          aria-hidden="true"
          class="fa fa-search"
        />
      </div>
       
      <div
        class="pr-2 pl-2 position-right-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
        style="display: none;"
      >
        <span
          class="loading-container text-center"
          style="display: none;"
        >
          <span
            aria-hidden="true"
            aria-label="Loading"
            class="spinner"
          />
        </span>
         
        <i
          aria-hidden="true"
          class="fa fa-times fa-lg cursor-pointer"
          data-original-title="Clear"
          style="display: none;"
          title=""
        />
      </div>
    </div>
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: foo
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: baz
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: qux
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: quux
        </span>
      </div>
    </li>
  </ul>
   
  <ul
    aria-label="Pagination"
    class="pagination gl-pagination d-flex justify-content-center prepend-top-default"
    role="navigation"
  >
    <li
      class="page-item disabled"
    >
      <span
        aria-disabled="true"
        aria-label="Go to previous page"
        class="page-link"
        href="#"
      >
        ‹ Prev
      </span>
    </li>
    <li
      class="page-item"
    >
      <a
        aria-label="Go to page 1"
        class="page-link  active btn-primary"
        href="#"
      >
        1
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        aria-label="Go to page 2"
        class="page-link"
        href="#"
      >
        2
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        aria-label="Go to page 3"
        class="page-link"
        href="#"
      >
        3
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        aria-label="Go to next page"
        class="page-link"
        href="#"
      >
        Next ›
      </a>
    </li>
  </ul>
   
  <!---->
</div>
`;

exports[`Paginated List Pagination renders 10 items for a default page size of 10, 13 total and renders pagination 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between flex-row-reverse"
  >
     
    <div
      class="position-relative ms-no-clear"
    >
      <input
        aria-label="Search"
        class="pl-5 form-control"
        placeholder="Search"
        type="text"
      />
       
      <div
        class="pl-2 pr-2 position-left-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
      >
        <i
          aria-hidden="true"
          class="fa fa-search"
        />
      </div>
       
      <div
        class="pr-2 pl-2 position-right-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
        style="display: none;"
      >
        <span
          class="loading-container text-center"
          style="display: none;"
        >
          <span
            aria-hidden="true"
            aria-label="Loading"
            class="spinner"
          />
        </span>
         
        <i
          aria-hidden="true"
          class="fa fa-times fa-lg cursor-pointer"
          data-original-title="Clear"
          style="display: none;"
          title=""
        />
      </div>
    </div>
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: foo
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: baz
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: qux
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: quux
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: corge
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: grault
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: garply
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: waldo
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: fred
        </span>
      </div>
    </li>
  </ul>
   
  <ul
    aria-label="Pagination"
    class="pagination gl-pagination d-flex justify-content-center prepend-top-default"
    role="navigation"
  >
    <li
      class="page-item disabled"
    >
      <span
        aria-disabled="true"
        aria-label="Go to previous page"
        class="page-link"
        href="#"
      >
        ‹ Prev
      </span>
    </li>
    <li
      class="page-item"
    >
      <a
        aria-label="Go to page 1"
        class="page-link  active btn-primary"
        href="#"
      >
        1
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        aria-label="Go to page 2"
        class="page-link"
        href="#"
      >
        2
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        aria-label="Go to next page"
        class="page-link"
        href="#"
      >
        Next ›
      </a>
    </li>
  </ul>
   
  <!---->
</div>
`;

exports[`Paginated List Pagination renders 13 items for a default page size of 20, 13 total items, and does not render pagination 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between flex-row-reverse"
  >
     
    <div
      class="position-relative ms-no-clear"
    >
      <input
        aria-label="Search"
        class="pl-5 form-control"
        placeholder="Search"
        type="text"
      />
       
      <div
        class="pl-2 pr-2 position-left-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
      >
        <i
          aria-hidden="true"
          class="fa fa-search"
        />
      </div>
       
      <div
        class="pr-2 pl-2 position-right-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
        style="display: none;"
      >
        <span
          class="loading-container text-center"
          style="display: none;"
        >
          <span
            aria-hidden="true"
            aria-label="Loading"
            class="spinner"
          />
        </span>
         
        <i
          aria-hidden="true"
          class="fa fa-times fa-lg cursor-pointer"
          data-original-title="Clear"
          style="display: none;"
          title=""
        />
      </div>
    </div>
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: foo
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: baz
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: qux
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: quux
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: corge
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: grault
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: garply
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: waldo
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: fred
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: xyzzy
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: plugh
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: thud
        </span>
      </div>
    </li>
  </ul>
   
  <!---->
   
  <!---->
</div>
`;

exports[`Paginated List Search states renders the custom empty search message when there are no search results 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between flex-row-reverse"
  >
     
    <div
      class="position-relative ms-no-clear"
    >
      <input
        aria-label="Search"
        class="pl-5 form-control pr-5"
        id="__BVID__42"
        placeholder="Search"
        type="text"
      />
       
      <div
        class="pl-2 pr-2 position-left-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
      >
        <i
          aria-hidden="true"
          class="fa fa-search"
        />
      </div>
       
      <div
        class="pr-2 pl-2 position-right-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
        style=""
      >
        <span
          class="loading-container text-center"
          style="display: none;"
        >
          <span
            aria-hidden="true"
            aria-label="Loading"
            class="spinner"
          />
        </span>
         
        <i
          aria-hidden="true"
          class="fa fa-times fa-lg cursor-pointer"
          data-original-title="Clear"
          style=""
          title=""
        />
      </div>
    </div>
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  />
   
  <!---->
   
  <div
    class="bs-callout bs-callout-warning mt-3 empty-message empty-search"
  >
    Custom empty message
  </div>
</div>
`;

exports[`Paginated List Search states renders the default empty search message when there are no search results 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between flex-row-reverse"
  >
     
    <div
      class="position-relative ms-no-clear"
    >
      <input
        aria-label="Search"
        class="pl-5 form-control pr-5"
        id="__BVID__26"
        placeholder="Search"
        type="text"
      />
       
      <div
        class="pl-2 pr-2 position-left-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
      >
        <i
          aria-hidden="true"
          class="fa fa-search"
        />
      </div>
       
      <div
        class="pr-2 pl-2 position-right-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
        style=""
      >
        <span
          class="loading-container text-center"
          style="display: none;"
        >
          <span
            aria-hidden="true"
            aria-label="Loading"
            class="spinner"
          />
        </span>
         
        <i
          aria-hidden="true"
          class="fa fa-times fa-lg cursor-pointer"
          data-original-title="Clear"
          style=""
          title=""
        />
      </div>
    </div>
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  />
   
  <!---->
   
  <div
    class="bs-callout bs-callout-warning mt-3 empty-message empty-search"
  >
    Sorry, your filter produced no results.
  </div>
</div>
`;

exports[`Paginated List Search states renders the list flitered by search results 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between flex-row-reverse"
  >
     
    <div
      class="position-relative ms-no-clear"
    >
      <input
        aria-label="Search"
        class="pl-5 form-control pr-5"
        id="__BVID__18"
        placeholder="Search"
        type="text"
      />
       
      <div
        class="pl-2 pr-2 position-left-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
      >
        <i
          aria-hidden="true"
          class="fa fa-search"
        />
      </div>
       
      <div
        class="pr-2 pl-2 position-right-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
        style=""
      >
        <span
          class="loading-container text-center"
          style="display: none;"
        >
          <span
            aria-hidden="true"
            aria-label="Loading"
            class="spinner"
          />
        </span>
         
        <i
          aria-hidden="true"
          class="fa fa-times fa-lg cursor-pointer"
          data-original-title="Clear"
          style=""
          title=""
        />
      </div>
    </div>
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: baz
        </span>
      </div>
    </li>
  </ul>
   
  <!---->
   
  <!---->
</div>
`;

exports[`Paginated List Searchless states renders the list in an empty state with a custom empty message, when one is provided 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between flex-row-reverse"
  >
     
    <gl-search-box-by-type-stub
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  />
   
  <!---->
   
  <div
    class="bs-callout bs-callout-warning mt-3 empty-message empty-message"
  >
    List is passed in empty
  </div>
</div>
`;

exports[`Paginated List Searchless states renders the list in an empty state, when no items are provided 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between flex-row-reverse"
  >
     
    <gl-search-box-by-type-stub
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  />
   
  <!---->
   
  <div
    class="bs-callout bs-callout-warning mt-3 empty-message empty-message"
  >
    There are currently no items in this list.
  </div>
</div>
`;

exports[`Paginated List Searchless states renders the list with nested items, when an array of items is provided 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between flex-row-reverse"
  >
     
    <gl-search-box-by-type-stub
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: foo
        </span>
      </div>
    </li>
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
  </ul>
   
  <gl-pagination-stub
    align="left"
    class="d-flex justify-content-center prepend-top-default"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnextpage="Go to next page"
    labelpage="page => \`Go to page \${page}\`"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next ›"
    perpage="10"
    prevtext="‹ Prev"
    totalitems="2"
    value="1"
  />
   
  <!---->
</div>
`;

exports[`Paginated List props filter filters on default "id" key 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between flex-row-reverse"
  >
     
    <gl-search-box-by-type-stub
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
  </ul>
   
  <gl-pagination-stub
    align="left"
    class="d-flex justify-content-center prepend-top-default"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnextpage="Go to next page"
    labelpage="page => \`Go to page \${page}\`"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next ›"
    perpage="10"
    prevtext="‹ Prev"
    totalitems="1"
    value="1"
  />
   
  <!---->
</div>
`;

exports[`Paginated List props filter filters on provided "myKey" key 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between flex-row-reverse"
  >
     
    <gl-search-box-by-type-stub
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: 
        </span>
      </div>
    </li>
  </ul>
   
  <gl-pagination-stub
    align="left"
    class="d-flex justify-content-center prepend-top-default"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnextpage="Go to next page"
    labelpage="page => \`Go to page \${page}\`"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next ›"
    perpage="10"
    prevtext="‹ Prev"
    totalitems="1"
    value="1"
  />
   
  <!---->
</div>
`;

exports[`Paginated List props filter filters with provided filter function 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between flex-row-reverse"
  >
     
    <gl-search-box-by-type-stub
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li
      class="list-group-item"
    >
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
  </ul>
   
  <gl-pagination-stub
    align="left"
    class="d-flex justify-content-center prepend-top-default"
    ellipsistext="…"
    labelfirstpage="Go to first page"
    labellastpage="Go to last page"
    labelnextpage="Go to next page"
    labelpage="page => \`Go to page \${page}\`"
    labelprevpage="Go to previous page"
    limits="[object Object]"
    nexttext="Next ›"
    perpage="10"
    prevtext="‹ Prev"
    totalitems="1"
    value="1"
  />
   
  <!---->
</div>
`;
